core/src/docs/upgrade.md

```diff
@@ -1,3 +1,18 @@
+# Upgrade to v0.33
+
+OpenDAL 0.33 redesigned the `Writer` API.
+
+- All `writer.append()` should be replaced by `writer.write()` instead.
+
+Please refer to [`Writer`](crate::Writer) for more details.
+
+And internally, services should handle `OpWrite::content_length` correctly.
+
+- If writer doesn't support upload unsized data, please return `NotSupported` if `content_length` is `None`.
```

Comment:
Seems that those are not the information for users. I don't think they should be here. Maybe need to write more from the user's perspective?

Reply:
> Maybe need to write more from the user's perspective?

Nice catch. OpenDAL has two kinds of users: public API user and raw API user.

Let me polis this part a bit.

Reply:
PTAL.